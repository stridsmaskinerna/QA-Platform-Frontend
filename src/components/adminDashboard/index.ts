export * from "./courseManagement";
export * from "./tagManagement";
export * from "./userManagement";
