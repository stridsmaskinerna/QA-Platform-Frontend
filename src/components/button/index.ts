export * from "./cancelButton";
export * from "./deleteButton";
export * from "./goBackButton";
export * from "./saveButton";
export * from "./okButton";
export * from "./editButton";
