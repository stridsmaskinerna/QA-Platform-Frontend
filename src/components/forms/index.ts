export * from "./LoginForm";
export * from "./RegisterForm";
export * from "./questionForms";
export * from "./answerForm";
export * from "./commentForm";
