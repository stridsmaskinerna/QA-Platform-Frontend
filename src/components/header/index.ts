export * from "./Header";
export * from "./NavMenu";
