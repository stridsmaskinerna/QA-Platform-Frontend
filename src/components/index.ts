export * from "./authGuard";
export * from "./header";
export * from "./errorBoundary";
export * from "./tabs";
export * from "./questionCard";
export * from "./forms";
export * from "./input";
export * from "./loader";
export * from "./search";
export * from "./questionFinder";
export * from "./modal";
export * from "./questionCardList";
export * from "./myQASection";
export * from "./select";
export * from "./richText";
export * from "./questionDetailsViewer";
export * from "./button";
export * from "./teacherDashboard";
export * from "./text";
