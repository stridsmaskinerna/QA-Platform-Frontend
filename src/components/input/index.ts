export * from "./Input";
export * from "./InputWithPrefetchedSuggestions";
