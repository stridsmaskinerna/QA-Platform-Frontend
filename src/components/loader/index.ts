export * from "./FullScreenLoader";
export * from "./Loader";
