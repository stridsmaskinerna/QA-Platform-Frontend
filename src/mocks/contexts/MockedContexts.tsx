import { ReactElement } from "react";
import { AuthProvider, LoaderProvider } from "../../context";
import { BrowserRouter } from "react-router";
import { I18nextProvider } from "react-i18next";
import i18n from "../../i18n";

interface IMockedContextsProps {
    children: ReactElement;
}

function MockedContexts({ children }: IMockedContextsProps) {
    return (
        <LoaderProvider>
            <AuthProvider>
                <I18nextProvider i18n={i18n}>
                    <BrowserRouter>{children}</BrowserRouter>
                </I18nextProvider>
            </AuthProvider>
        </LoaderProvider>
    );
}

export default MockedContexts;
