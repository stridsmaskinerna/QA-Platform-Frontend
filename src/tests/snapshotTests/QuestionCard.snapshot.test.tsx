import { expect, it, vi } from "vitest";
import { QuestionCard } from "../../components";
import { exampleQuestionCardProps } from "../testUtils";
import { render } from "@testing-library/react";
import MockedContexts from "../../mocks/contexts/MockedContexts";

it("QuestionCard to match snapshot", () => {
    const { asFragment } = render(
        <MockedContexts>
            <QuestionCard
                isPostedByUser={true}
                data={exampleQuestionCardProps}
                handleDeleteClick={vi.fn}
            />
        </MockedContexts>,
    );
    expect(asFragment()).toMatchSnapshot();
});
