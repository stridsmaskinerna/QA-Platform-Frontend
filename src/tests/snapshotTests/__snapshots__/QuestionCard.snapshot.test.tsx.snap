// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`QuestionCard to match snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <div
      class="container"
    >
      <h3
        class="courseTitle"
        data-hl="title"
      >
        007 - subjectName
      </h3>
      <div
        class="iconSection"
      >
        <div
          class="resolvedContainer resolved"
        >
          <img
            alt="Resolved"
            class="icon"
            src="data:image/svg+xml,%3csvg%20xmlns='http://www.w3.org/2000/svg'%20height='24px'%20viewBox='0%20-960%20960%20960'%20width='24px'%20fill='%23FFFFFF'%3e%3cpath%20d='m424-296%20282-282-56-56-226%20226-114-114-56%2056%20170%20170Zm56%20216q-83%200-156-31.5T197-197q-54-54-85.5-127T80-480q0-83%2031.5-156T197-763q54-54%20127-85.5T480-880q83%200%20156%2031.5T763-763q54%2054%2085.5%20127T880-480q0%2083-31.5%20156T763-197q-54%2054-127%2085.5T480-80Zm0-80q134%200%20227-93t93-227q0-134-93-227t-227-93q-134%200-227%2093t-93%20227q0%20134%2093%20227t227%2093Zm0-320Z'/%3e%3c/svg%3e"
          />
          <span
            class="resolvedTextContainer"
            data-hl="resolved-question"
          >
            Resolved
          </span>
        </div>
        <div
          class="iconWrapper"
        >
          <img
            alt=""
            class="icon"
            data-hl="public-question"
            src="data:image/svg+xml,%3csvg%20xmlns='http://www.w3.org/2000/svg'%20height='24px'%20viewBox='0%20-960%20960%20960'%20width='24px'%20fill='%23FFFFFF'%3e%3cpath%20d='M480-80q-83%200-156-31.5T197-197q-54-54-85.5-127T80-480q0-83%2031.5-156T197-763q54-54%20127-85.5T480-880q83%200%20156%2031.5T763-763q54%2054%2085.5%20127T880-480q0%2083-31.5%20156T763-197q-54%2054-127%2085.5T480-80Zm-40-82v-78q-33%200-56.5-23.5T360-320v-40L168-552q-3%2018-5.5%2036t-2.5%2036q0%20121%2079.5%20212T440-162Zm276-102q41-45%2062.5-100.5T800-480q0-98-54.5-179T600-776v16q0%2033-23.5%2056.5T520-680h-80v80q0%2017-11.5%2028.5T400-560h-80v80h240q17%200%2028.5%2011.5T600-440v120h40q26%200%2047%2015.5t29%2040.5Z'/%3e%3c/svg%3e"
          />
          <span
            class="tooltip"
            data-hl="public-question-span"
          >
            Public question
          </span>
        </div>
        <div
          class="iconWrapper"
        >
          <button
            class="hideShowBtn"
          >
            <img
              alt=""
              class="icon"
              src="data:image/svg+xml,%3csvg%20xmlns='http://www.w3.org/2000/svg'%20height='24px'%20viewBox='0%20-960%20960%20960'%20width='24px'%20fill='%23FFFFFF'%3e%3cpath%20d='M480-320q75%200%20127.5-52.5T660-500q0-75-52.5-127.5T480-680q-75%200-127.5%2052.5T300-500q0%2075%2052.5%20127.5T480-320Zm0-72q-45%200-76.5-31.5T372-500q0-45%2031.5-76.5T480-608q45%200%2076.5%2031.5T588-500q0%2045-31.5%2076.5T480-392Zm0%20192q-146%200-266-81.5T40-500q54-137%20174-218.5T480-800q146%200%20266%2081.5T920-500q-54%20137-174%20218.5T480-200Zm0-300Zm0%20220q113%200%20207.5-59.5T832-500q-50-101-144.5-160.5T480-720q-113%200-207.5%2059.5T128-500q50%20101%20144.5%20160.5T480-280Z'/%3e%3c/svg%3e"
              title="Set question to hidden"
            />
          </button>
        </div>
      </div>
    </div>
    <a
      class="link"
      data-discover="true"
      href="/guest/questions/1"
    >
      <div
        class="container"
      >
        <div
          class="header"
        >
          <span
            class="topicName"
            data-hl="topic-name"
          >
            topicName
          </span>
          <div
            class="timeSection"
          >
            <img
              alt=""
              class="icon"
              src="data:image/svg+xml,%3csvg%20xmlns='http://www.w3.org/2000/svg'%20height='24px'%20viewBox='0%20-960%20960%20960'%20width='24px'%20fill='%23808080'%3e%3cpath%20d='m612-292%2056-56-148-148v-184h-80v216l172%20172ZM480-80q-83%200-156-31.5T197-197q-54-54-85.5-127T80-480q0-83%2031.5-156T197-763q54-54%20127-85.5T480-880q83%200%20156%2031.5T763-763q54%2054%2085.5%20127T880-480q0%2083-31.5%20156T763-197q-54%2054-127%2085.5T480-80Zm0-400Zm0%20320q133%200%20226.5-93.5T800-480q0-133-93.5-226.5T480-800q-133%200-226.5%2093.5T160-480q0%20133%2093.5%20226.5T480-160Z'/%3e%3c/svg%3e"
            />
            <div>
              <p
                data-hl="creation-date"
              >
                Less than a minute ago
              </p>
            </div>
          </div>
        </div>
        <div
          class="questionSection"
        >
          <h2
            class="title"
            data-hl="title-question"
          >
            title
          </h2>
          <p
            data-hl="userName"
          >
            Asked by: username
          </p>
          <p
            data-hl="answer-nount"
          >
            5 answers
          </p>
        </div>
      </div>
    </a>
    <div
      class="container"
    >
      <div
        class="tagsWrapper"
      >
        <p
          class="tagContainer"
          data-hl="tags"
        >
          TAG1, TAG2, TAG3, TAG4
        </p>
      </div>
      <div
        class="btnsContainer"
      >
        <button
          class="button"
        >
          Edit
        </button>
        <button
          class="button"
          type="button"
        >
          Delete
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
